### DISCUSSION TITLE
Illegal offset in edit extension

### DISCUSSION THREAD

**Discussion by neto737** [datenstrom/community#529] [Report a bug] [answered]
Hi,

I'm receiving an error when I try to create a new account or recover an account through email using the Edit extension.

Warning: Illegal offset type in /website.tld/public_html/system/extensions/core.php on line 954

It's my first time using this software, I don't know if the devs are already aware about that or if it's a problem in my side.

Thanks in advance.

**Comment by markseuffert** [comment]
Welcome, 

I can't reproduce the problem on my server. I downloaded the latest version and could create/recover an account. Can you show your `system/extensions/yellow.log`? Have any additional information what could cause this problem?

**Comment by neto737** [comment]
There's my yellow.log

```
2020-07-20 01:32:41 info Datenstrom Yellow 0.8.13, PHP 7.3.20-1+0~20200710.65+debian9~1.gbpc9cbeb, Apache 2.4.43, Linux
2020-07-20 01:32:41 info Install extension 'English 0.8.20'
2020-07-20 01:32:41 info Install extension 'German 0.8.20'
2020-07-20 01:32:41 info Install extension 'Portuguese 0.8.20'
2020-07-20 01:33:06 info Add user 'Neto'
```

Weirdly this problem occurs only in my domain .dev (that requires HSTS to work). I've created a subdomain in another domain I have and tried without using HTTPS and everything worked fine. Also in the same subdomain when I use HTTPS it stop to work. It seems to be a problem when I use HTTPS protocol.

I can provide more info privately (FTP to this domain, hosting details, etc) if you want to take a look at it.

**Comment by markseuffert** [comment]
Thanks for the information. I am a maintainer for Datenstrom Yellow. Please contact me and I have a closer look.

**Comment by markseuffert** [comment, marked as answer]
Thanks, I was able to reproduce the problem on your server. I fixed the PHP error and also updated the automatic URL detection. It now looks at `REQUEST_SCHEME` and `X-Forwarded-Proto` HTTP header.

Should be working now. Can you make a new installation and test again?

**Comment by neto737** [comment]
I've just tested it and everything seems to be working fine now. I really thank you for fixing it so fast, thanks.
